packages/react-scripts/config/config.js

```diff
@@ -0,0 +1,110 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const paths = require('./paths');
+
+const appDirectory = fs.realpathSync(process.cwd());
+
+/**
+ * Get the baseUrl of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getBaseUrl(options = {}) {
+  const baseUrl = options.baseUrl;
+
+  if (!baseUrl) {
+    return null;
+  }
+
+  if (baseUrl !== 'src' && baseUrl !== './src') {
+    console.error(
+      chalk.red.bold(
+        "You tried to set baseUrl to anything other than 'src'. This is not supported in create-react-app and will be ignored."
+      )
+    );
+
+    return null;
+  }
+
+  return path.resolve(appDirectory, 'src');
+}
+
+/**
+ * Get the alias of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getAlias(options = {}) {
+  const paths = options.paths || {};
+
+  const alias = paths['@'];
+
+  const others = Object.keys(paths).filter(function(value) {
+    return value !== '@';
+  });
+
+  if (others.length) {
+    console.error(
+      chalk.red.bold(
+        'You tried to set one or more paths with an alias other than "@", this is currently not supported in create-react-app and will be ignored.'
+      )
+    );
+  }
+
+  if (!alias) {
+    return {};
+  }
+
+  if (alias.toString() !== 'src') {
+    console.error(
+      chalk.red.bold(
+        "You tried to set a path with alias '@' to anything other than ['src']. This is not supported in create-react-app and will be ignored."
+      )
+    );
+  }
+
+  return {
+    '@': path.resolve(appDirectory, 'src'),
+  };
+}
+
+function getConfig() {
+  // Check if TypeScript is setup
+  const useTypeScript = fs.existsSync(paths.appTsConfig);
+  const hasJsConfig = fs.existsSync(paths.appJsConfig);
```

Comment:
We should probably enforce not allowing `jsconfig.json` and `tsconfig.json` both being present in a preflight check.

Reply:
Done!